lib/elixir/lib/system.ex

```diff
@@ -427,8 +427,8 @@ defmodule System do
   Returns the value of the given environment variable.
 
   The returned value of the environment variable
-  `varname` is a string, or `nil` if the environment
-  variable is not set.
+  `varname` is a string. If the environment variable
+  is undefined, returns default (or `nil` if not provided)
```

Comment:
Minor nitpick: use "not set" instead of "undefined" for not un-doing previous efforts https://github.com/elixir-lang/elixir/pull/8851

```suggestion
  is not set, returns `default` (which is `nil` unless specified 
  otherwise).
```

Reply:
We should also add in the documentation as well that the default needs to be a binary.

Reply:
wdyt:

>   The returned value of the environment variable
  `varname` is a string. If the environment variable
  is not set, returns the string specified in `default` or
 `nil` if none is set.

Reply:
I'm not sure what you mean @guilleiguaran, I'm sorry. I'm just saying that we are requiring the `default` argument to be either `nil` or a string but we're not documenting that if present, the default argument should be a string.

Reply:
@whatyouhide oops sorry, the format of the comment was wrong:

> The returned value of the environment variable
> varname is a string. If the environment variable
> is not set, returns the string specified in default or
> nil if not specified.

(maybe `none is specified` sounds better than `not specified`?)